pkg/operator/ceph/cluster/mon/spec.go

```diff
@@ -193,6 +194,11 @@ func (c *Cluster) makeMonPod(monConfig *monConfig) *v1.Pod {
 		Volumes:           opspec.DaemonVolumesBase(monConfig.DataPathMap, keyringStoreName),
 		HostNetwork:       c.Network.IsHost(),
 		PriorityClassName: cephv1.GetMonPriorityClassName(c.spec.PriorityClassNames),
+		Tolerations: []v1.Toleration{
```

Comment:
The mons can't be evicted if they are using dataDirHostPath. They can only be evicted if they are based on PVCs. So I think we only want the toleration for mons if backed by PVCs.

Reply:
Ok. I will see to change it for this case only. And i will check if it is possible also to add the Toleration to  mds and rgw pods